library/Mockery/Generator/Parameter.php

```diff
@@ -48,7 +48,13 @@ public function getTypeHintAsString()
         if ((version_compare(PHP_VERSION, '5.4.1') >= 0)) {
             try {
                 if ($this->rfp->getClass()) {
-                    return $this->rfp->getClass()->getName();
+                    $typehintedClass = $this->rfp->getClass()->getName();
+
+                    if ($this->isNamespacedClass($typehintedClass)) {
```

Comment:
Reflections always give you back FQCN class name without `\`. As you've noted with used in `eval` this ends up pretty badly by inheriting namespace of current class, where eval is made.